### ISSUE TITLE
Putting a lyluatex environment inside a macro

### ISSUE THREAD

**Issue by uliska** [jperon/lyluatex#191]
I'm experiencing a problem with what boils down to this MWE:

```TeX
\documentclass{scrartcl}
\usepackage{lyluatex}

\newcommand{\testcompile}{%
  \begin{lilypond}[]
    \engrave something
    \score {
      \new Staff { c' }
    }
  \end{lilypond}
}

\begin{document}

\testcompile

\end{document}
```

LilyPond will choke on `\engrave` (just for the sake of demonstration), and the special thing is that the `lilypond` block is wrapped in a macro.

Compiling this with *any* optional command in the `\begin{lilypond}[]` will make it fail with the following **LaTeX** error:

```
! Undefined control sequence.
\testcompile ->\begin {lilypond}[] \engrave 
                                            something \score { \new Staff { ...
```
*Removing* the square brackets the file successfully compiles and includes the score with the following - strange - console output:

```
Module lyluatex Warning: Found something incompatible with `fragment`
(lyluatex)               (or `relative`). Setting them to false.
(lyluatex)               on input line 15
```
... this is expected
```
(lyluatex)	Compiling score tmp-ly/903e41574fa322b52199b024994808f6 with LilyPond executable 'lilypond'.
-:30:1: Fehler: Ung?ltige Fluchtsequenz: >>\engrave<<

\engrave something \score { \new Staff { c' } }
-:30:10: Fehler: syntax error, unexpected SYMBOL, expecting '.' or '=' or ','
\engrave 
         something \score { \new Staff { c' } }
schwerer Fehler: gescheiterte Dateien: "-"
```
... this is strange: usually lyluatex does *not* print LilyPond's log directly
```
Module lyluatex Warning: 
(lyluatex)               LilyPond reported a failed compilation but
(lyluatex)               produced a score. If you need more information
(lyluatex)               than the above message,
(lyluatex)               please retry with option debug=true.
(lyluatex)               
(lyluatex)               on input line 15
```
This again is expected.

I know that `lilypond` behaves specially, and there's even a section for that in the manual, but I have the impression that 
* including `lilypond` in a *macro* behaves just the opposite from including in an *environment* - in so far as *no* square brackets are *allowed*
* even with no arguments it will work but it will behave differently by directly printing LilyPond's output to the console/log

I have found a workaround for my current project, but could you help me understand the issue, so we can update the manual section?

**Comment by jperon** [comment]
Simply put, it isn't possible to put the `ly` environment in a macro : you can only wrap `\lily` that way.

The reason is that `\begin{ly}` calls a special behavior of luatex, and the lua code needs to detect a line beginning with `\end` to leave that mode ; so it only works with environments.

I'll be more explicit on that point in the manual, then close this issue.

**Comment by jperon** [comment]
Fixed in c58940a60640d2dac897e10fd04bab5774e93fe1.
